abstractions/typescript/src/requestInfo.ts

```diff
@@ -15,6 +16,22 @@ export class RequestInfo {
     public queryParameters: Map<string, object> = new Map<string, object>(); //TODO: case insensitive
     /** The Request Headers. */
     public headers: Map<string, string> = new Map<string, string>(); //TODO: case insensitive
+    private _middlewareOptions = new Map<string, MiddlewareOption>(); //TODO: case insensitive
```

Comment:
Please use something like [MiddlewareControl](https://github.com/microsoftgraph/msgraph-sdk-javascript/blob/dev/src/middleware/MiddlewareControl.ts instead of using _middlewareOptions. Please move addMiddlewareOptions and getMiddlewareOptions options to the MiddlewareControl.


- A separate class for managing the properties as an object allows extensibility. A user can extend the MiddlewareControl class and flexibly customize their get and add middleware options or any property of the requestinfo class.
- We can observe Single Responsibility Principle if we neatly segregate some functionalities into specific classes.
- Avoids bulking up the RequestInfo class with too many methods.

Reply:
Yeah I was looking at the dotnet sdk which uses options but have uses control too. I'm not sure why we have a difference between languages today. @darrelmiller any guidance on options or control?
https://www.github.com/microsoftgraph/msgraph-sdk-dotnet-core/tree/dev/src%2FMicrosoft.Graph.Core%2FRequests%2FMiddleware%2FOptions%2FIMiddlewareOption.cs

We're attaching the options to request info  to avoid passing multiple parameters between the layers. Request info is already being passed between layers. And the options or controls are related to the request info, which is why it felt natural to add them there. I'm not sure how you'd do it differently without having to pass additional parameters to the http core service?

Reply:
Both JS and .NET use the term Options.  It looks like JS has a container for for middlewareOptions.  .NET just uses a dictionary.  I'm not sure why JS has the extra layer of indirection.  Part of the confusion might be because the original requirement document talked about middleware controls.  That notion got renamed to middleware options during implementation.

Reply:
Thanks for the context here. So I'm guessing we're sticking with options then.